docs/12.EXT-REFERENCE-MODULE.md

```diff
@@ -13,14 +13,15 @@ type of module is requested.
 
 Additionally, this extension provides a means of easily defining so-called "native" JerryScript modules which can be
 resolved using the JerryScript native module resolver `jerryx_module_native_resolver()`, which can be passed to
-`jerryx_module_resolve()`. Note, however, that native JerryScript modules are only supported and
-`jerryx_module_native_resolver()` is only compiled in if compiler support for `__attribute__` extensions is present. In
-effect this means that native JerryScript modules are available only when this extension is built with GCC or
-LLVM/clang. In the absence of such support, you may construct alternative module systems and provide your own resolver
-to `jerryx_module_resolve()`.
+`jerryx_module_resolve()`. Note, however, that automatic loading of native JerryScript modules is only supported if
+compiler support for `__attribute__` extensions is present. In effect this means that automatic loading for native
+JerryScript modules is available only when this extension is built with GCC or LLVM/clang. In the absence of such
+support, you may manually initialize the list of native modules recognized by the native resolver by passing an array
+of `jerryx_native_module_t` structures to the macro `JERRYX_NATIVE_MODULES_INIT()` preferably during execution of
+`main()`. Note that it is not necessary to call `jerry_init()` before calling the macro.
 
-`jerryscript-ext/module.h` defines the preprocessor directive `JERRYX_NATIVE_MODULES_SUPPORTED` only if support for
-native JerryScript modules is available.
+`jerryscript-ext/module.h` defines the preprocessor directive `JERRYX_NATIVE_MODULES_SUPPORTED` and sets it to `1` only
+if compiler support for automatic loading of native JerryScript modules is available.
```

Comment:
We may explicitly define JERRYX_NATIVE_MODULES_SUPPORTED as non-zero when available since the zero check determines that it is not available.

Reply:
I actually want it to be conditionally defined in the header file, because I want the build system to be able to override it so, for example, I can define it and set it to zero so I can tests the implementation without linker sections even on gcc, where it would normally be defined and set to 1.